Validator Vesting Upgrade v44 + Cosmos x/auth migration

DracoLi opened the pull request:
- Upgrades validator-vesting to v44. Since all validator vesting accounts should be migrated to a base account now, I removed most of validator-vesting logic aside from client stuff since we use it for supply related queries.
- Cosmos x/auth migration can now be added now that validator vesting is migrated to v44. I've included that + some test cases to make sure the accounts are properly migrated.

The `appstate-auth-v15.json` contains both auth, supply, and bank app states since v44 moved supply and auth coins to the bank module. The tests here should cover those and it seems to be correctly migrated.

drklee3 commented on x/validator-vesting/client/cli/query.go:
```diff
@@ -5,212 +5,237 @@ import (
 		Args:  cobra.NoArgs,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			cliCtx := context.NewCLIContext().WithCodec(cdc)
+			cliCtx, err := client.GetClientTxContext(cmd)
```
```suggestion
			cliCtx, err := client.GetClientQueryContext(cmd)
```

I switched these up a lot before too, same for the other query commands

rhuairahrighairidh commented on migrate/v0_16/testdata/appstate-bank-v15.json:
```diff
@@ -0,0 +1,541 @@
+            }
+          ],
+          "public_key": null,
+          "sequence": "0"
```
nit: all the `sequence` values are 0 which can hide bugs. If the migrations got the tag name wrong they'd look for a nonexistent field, and fill in the value with the default 0. I'd maybe change a couple to something other than 0.

rhuairahrighairidh commented on migrate/v0_16/testdata/appstate-bank-v15.json:
```diff
@@ -0,0 +1,541 @@
+              "denom": "ukava"
+            }
+          ],
+          "public_key": null,
```
nit: All the public_key's are empty, I'd add some to test.

rhuairahrighairidh reviewed (approved):
There's a rogue print statement, but otherwise looks good.

I think a test running on a full mainnet export would be good to sanity check account balances and no panics. But that can be done later, I added a card.

DracoLi commented on migrate/v0_16/testdata/appstate-bank-v15.json:
```diff
@@ -0,0 +1,541 @@
+        "value": {
+          "address": "kava1cj7njkw2g9fqx4e768zc75dp9sks8u9znxrf0w",
+          "coins": [
+            {
+              "denom": "ukava",
+              "amount": "1000000000000"
+            },
+            {
+              "denom": "hard",
+              "amount": "1000000000000"
+            },
+            {
+              "denom": "swp",
+              "amount": "1000000000000"
+            }
```
👍

DracoLi commented:
@rhuairahrighairidh thx for the review. addressed feedbacks here https://github.com/Kava-Labs/kava/pull/1072/commits/7a124ad76861fb0105834458db1412c9dbbb50fb
